File: c/leetcode/53. Maximum Subarray.c


int maxSubArray(int *nums, int numsSize)
{
  if (!numsSize)
    return 0;
  int max = nums[0];
  int sum = max;

  for (int i = 1; i < numsSize; i++)
  {
    sum = nums[i] > nums[i] + sum ? nums[i] : nums[i] + sum;
    max = max > sum ? max : sum;
  }

  return max;
}

File: javascript/leetcode/53. Maximum Subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
  if (!nums.length) return 0;

  let max = nums[0];
  let sum = max;

  nums.forEach((el, ind) => {
    if (ind !== 0) {
      sum = Math.max(el, el + sum);
      max = max > sum ? max : sum;
    }
  });

  return max;
};
